### PR TITLE
Closes #443

### LINKED ISSUE #443: test: no E2E coverage for Zip64 boundary (REQ_E-015: >65,535 files / >4GB)
## Severity: High (flagship requirement unverified end-to-end)

REQ_E-015 mandates archives exceeding **4GB and 65,535 Native Files (Zip64)**. The E2E suite never crosses this boundary:

- Max `--count` anywhere in `tests/*.sh` is **1000**.
- No archive approaches 4GB.
- No assertion references the Zip64 code path (`grep -rniE 'zip64|65535|65536|4294967296' tests/` → no matches).

A regression in the Zip64 central-directory / EOCD64 path (entry count rollover at 65,535, offset rollover at 4GB) would ship undetected. This is the product's headline capability.

Note: #436 (high-volume smoke at `--count 5000`) targets fd/OOM limits, **not** the Zip64 entry-count or size boundary — distinct concern.

## Proposed coverage (cheap variant)

A separate/nightly E2E case (not in the fast inner loop):

1. `--count 70000` with the smallest possible native (e.g. empty/1-byte placeholder) → crosses the 65,535 entry boundary.
2. Assert entry count == 70000 in the archive.
3. `unzip -t` (or `[System.IO.Compression.ZipFile]::OpenRead` enumerate) to confirm the archive is structurally valid and Zip64-readable by a second tool.
4. Optionally a `--target-zip-size` >4GB case behind an opt-in flag/CI label (slow, disk-heavy) to exercise the 4GB offset path.

Keep it out of the per-PR critical path; gate behind a label or schedule. The 70k-entry tiny-file case is fast and catches the more likely (count-boundary) regression.

Source: senior-QA E2E coverage review (follow-on to #360 / #363–#367).

### PR DESCRIPTION
**Pull request by dwojtaszek**
Added E2E coverage for Zip64 boundary. Includes nightly workflow for automated validation.

## Summary by CodeRabbit

* **New Features**
  * Added ZIP file size reporting and verification to validate generated archives against target sizes.
  * Introduced automated nightly testing for Zip64 boundary conditions and large file handling.

* **Chores**
  * Improved build reproducibility with deterministic builds configuration.
  * Enhanced CI/CD pipeline validation and testing infrastructure.

* **Tests**
  * Added comprehensive test coverage for size verification and CI build properties validation.

### CODE REVIEW

File: src/ZipSizeVerifier.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+namespace Zipper
+{
+    internal static class ZipSizeVerifier
+    {
+        public static (bool IsWithinTolerance, double Deviation) Verify(long targetSize, long actualSize)
+        {
+            double deviation = targetSize > 0 ? Math.Abs(actualSize - targetSize) / (double)targetSize : 0;
+            bool isWithinTolerance = deviation <= 0.10;
+            return (isWithinTolerance, deviation);
+        }
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/ZipSizeVerifier.cs]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+mkdir -p "$TEST_OUTPUT_DIR"
+
+COUNT=70000
+
+print_info "Generating archive with $COUNT files to cross 65,535 boundary..."
+
+# Use zipper command defined in _zipper-cli.sh
+zipper --type jpg --count "$COUNT" --output-path "$TEST_OUTPUT_DIR" --seed 42 
+
+ZIP_FILE=$(find "$TEST_OUTPUT_DIR" -name "*.zip" -print -quit)
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+# 5. >4GB target size case (Behind run_4gb flag)
+if [[ "${RUN_4GB_CASE:-false}" == "true" ]]; then
+    print_info "Running 4GB target size Zip64 boundary test..."
+    OUT_4GB="$TEST_OUTPUT_DIR/4gb"
+    mkdir -p "$OUT_4GB"
+    
+    zipper --type pdf --count 10 --target-zip-size 4500MB --output-path "$OUT_4GB" --seed 42 
+    
+    ZIP_4GB=$(find "$OUT_4GB" -name "*.zip" -print -quit)
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: src/Zipper.Tests/ZipSizeVerifierTests.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        [Theory]
+        // Exact boundaries (inclusive/exclusive)
+        [InlineData(1000, 1000, true, 0.0)]
+        [InlineData(1000, 1100, true, 0.10)]
+        [InlineData(1000, 900, true, 0.10)]
+        [InlineData(1000, 1101, false, 0.101)]
+        [InlineData(1000, 899, false, 0.101)]
+        // Zero target
+        [InlineData(0, 1000, true, 0.0)]
+        [InlineData(0, 0, true, 0.0)]
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/Zipper.Tests/ZipSizeVerifierTests.cs]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+if [[ -z "$ZIP_FILE" ]]; then
+    print_error "No zip file generated"
+    exit 1
+fi
+
+print_info "Found generated archive: $ZIP_FILE"
+
+# 1. Entry count assertions using unzip -Z1
+JPG_COUNT=$(unzip -Z1 "$ZIP_FILE" | grep -cE '\.jpg$') || true
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    mkdir -p "$OUT_4GB"
+    
+    zipper --type pdf --count 10 --target-zip-size 4500MB --output-path "$OUT_4GB" --seed 42 
+    
+    ZIP_4GB=$(find "$OUT_4GB" -name "*.zip" -print -quit)
+    if [[ -z "$ZIP_4GB" ]]; then
+        print_error "No 4GB zip file generated"
+        FAILED=$((FAILED + 1))
+    else
+        SIZE=$(stat -c%s "$ZIP_4GB" 2>/dev/null || stat -f%z "$ZIP_4GB")
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    FAILED=$((FAILED + 1))
+fi
+
+# 4. Load-file row count (.opt file generated alongside)
+DAT_FILE="${ZIP_FILE%.zip}.opt"
+
+if [[ -f "$DAT_FILE" ]]; then
+    # Load files have a header row, so 70000 files = 70001 lines
+    DAT_LINES=$(wc -l < "$DAT_FILE" | xargs)
+    if [[ "$DAT_LINES" -eq 70000 ]]; then
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: src/Zipper.Tests/ColumnProfileLoaderTests.cs [hunk 8]
```diff
@@ -504,6 +504,7 @@ public void Validate_WithLegacyColumnKind_DoesNotThrow()
         [Fact]
         public void Validate_WithNonUsCulture_ParsesIsoDateRangesCorrectly()
         {
+#pragma warning disable RS0030 // Do not use banned APIs
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/Zipper.Tests/ColumnProfileLoaderTests.cs]: We are removing the method-scoped pragmas to rely on the file-scoped suppression instead. Tests run sequentially so global mutation is acceptable for now.

File: src/Zipper.Tests/Profiles/Generation/DateGeneratorTests.cs [hunk 9]
```diff
@@ -29,6 +29,7 @@ public class DateGeneratorTests
     [Fact]
     public void DateGenerator_WithNonUsCulture_ParsesIsoDatesCorrectly()
     {
+#pragma warning disable RS0030 // Do not use banned APIs
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/Zipper.Tests/Profiles/Generation/DateGeneratorTests.cs]: We are removing the method-scoped pragmas to rely on the file-scoped suppression instead. Tests run sequentially so global mutation is acceptable for now.

File: .github/workflows/zip64-nightly.yml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        default: false
+
+permissions:
+  issues: write
+
+jobs:
+  test-zip64-boundary:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply .github/workflows/zip64-nightly.yml]: Fixed in the latest commit.

File: src/ZipSizeVerifier.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+namespace Zipper
+{
+    internal static class ZipSizeVerifier
+    {
+        public static (bool IsWithinTolerance, double Deviation) Verify(long targetSize, long actualSize)
+        {
+            double deviation = targetSize > 0 ? Math.Abs(actualSize - targetSize) / (double)targetSize : 0;
+            bool isWithinTolerance = deviation <= 0.10;
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/ZipSizeVerifier.cs]: Fixed in the latest commit.

File: tests/test-ci-build-props.sh [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+echo "Testing ContinuousIntegrationBuild property (GITHUB_ACTIONS=true)..."
+GH_BUILD=$(GITHUB_ACTIONS=true dotnet msbuild ../src/Zipper.csproj -getProperty:ContinuousIntegrationBuild)
+if [[ "$GH_BUILD" != "true" ]]; then
+    echo "FAIL: ContinuousIntegrationBuild is not true when GITHUB_ACTIONS=true. Got '$GH_BUILD'"
+    exit 1
+fi
+
+echo "Testing ContinuousIntegrationBuild property (no CI)..."
+NO_CI_BUILD=$(CI= GITHUB_ACTIONS= dotnet msbuild ../src/Zipper.csproj -getProperty:ContinuousIntegrationBuild)
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-ci-build-props.sh]: Skipping: The script uses cd $(dirname $0) at the top, so the relative path is robust to the invocation location.

File: tests/test-zip64-boundary.sh [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+        print_info "Assertion OK: Load file (.opt) has 70000 lines"
+        PASSED=$((PASSED + 1))
+    else
+        print_error "Assertion FAILED: Load file has $DAT_LINES lines, expected 70000"
+        FAILED=$((FAILED + 1))
```
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh:87]: Fixed in the latest commit.

File: src/ZipSizeVerifier.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/ZipSizeVerifier.cs]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: src/Zipper.Tests/ZipSizeVerifierTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/Zipper.Tests/ZipSizeVerifierTests.cs]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: tests/test-zip64-boundary.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh]: Fixed in the latest commit.

File: src/Zipper.Tests/ColumnProfileLoaderTests.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/Zipper.Tests/ColumnProfileLoaderTests.cs]: We are removing the method-scoped pragmas to rely on the file-scoped suppression instead. Tests run sequentially so global mutation is acceptable for now.

File: src/Zipper.Tests/Profiles/Generation/DateGeneratorTests.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/Zipper.Tests/Profiles/Generation/DateGeneratorTests.cs]: We are removing the method-scoped pragmas to rely on the file-scoped suppression instead. Tests run sequentially so global mutation is acceptable for now.

File: .github/workflows/zip64-nightly.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply .github/workflows/zip64-nightly.yml]: Fixed in the latest commit.

File: src/ZipSizeVerifier.cs (hunk 11 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply src/ZipSizeVerifier.cs]: Fixed in the latest commit.

File: tests/test-ci-build-props.sh (hunk 12 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-ci-build-props.sh]: Skipping: The script uses cd $(dirname $0) at the top, so the relative path is robust to the invocation location.

File: tests/test-zip64-boundary.sh (hunk 13 above)
[the remark this answers is not in the document]
Comment (dwojtaszek) [reply tests/test-zip64-boundary.sh:87]: Fixed in the latest commit.

**Comment by dwojtaszek** [comment]
I have fixed the pipeline masking issues in tests/test-zip64-boundary.sh (using `|| true` on the find command). I also resolved the merge conflicts with main, keeping the file-scoped namespace for ZipSizeVerifier, using main's isZeroMatch logic, and adopting main's invariant globalization approach (which means the flaky CultureInfo tests were removed as in main).
